packages/hash/api/src/model/entityType.model.ts

```diff
@@ -234,13 +234,25 @@ class __EntityType {
     return dbEntityTypes.map((entityType) => new EntityType(entityType));
   }
 
-  static async getEntityTypeParents(
-    client: DBClient,
-    params: { entityTypeId: string },
-  ) {
-    const dbEntityTypes = await client.getEntityTypeParents(params);
+  async getParentEntityTypes(client: DBClient) {
+    const parentSchema$ids = (
+      Array.isArray(this.properties?.allOf) ? this.properties.allOf : []
+    )
+      .filter(
+        (allOfEntry: any): allOfEntry is { $ref: string } =>
+          "$ref" in allOfEntry,
+      )
+      .map((allOfEntry) => allOfEntry.$ref);
```

Comment:
Doesn't have to be done as part of this PR, but I think we should type `this.properties` as a JSON-Schema (using a package like `@types/json-schema` or equivalent). That would let you skip checking if `allOf` is an array.

I tried it out locally, and this actually also revealed is that a boolean value can be in the `allOf` array so an additional check is required: 
```suggestion
    const parentSchema$ids =
      this.properties?.allOf
        ?.filter(
          (allOfEntry): allOfEntry is { $ref: string } =>
            typeof allOfEntry === "object" && "$ref" in allOfEntry,
        )
        .map(({ $ref }) => $ref) ?? [];
```

Reply:
I appreciate we will likely want to extend the existing TS definition (for instance if we're enforcing every schema has certain fields, like an `allOf` array that is either empty or includes references) - but existing TS definitions would get us a long way now.

Reply:
> I tried it out locally, and this actually also revealed is that a boolean value can be in the allOf array so an additional check is required

Did you add the JSON schema typing locally? I would be interested in doing that to pick up errors such as this. Did you just add it as a dependency and imported the type?

Reply:
I added `json-schema` and `@types/json-schema` as dependencies, and used their `JSONSchema7` type - not sure if this is the best one out there though.

Reply:
I'll look around. We have specified [2019-09 for future schemas (asana task)](https://app.asana.com/0/1200211978612931/1201852312063660/f), but if libraries fail to supply types for that, we might need to reconsider

Reply:
Hmm that's a good point, I couldn't find a great alternative (there is [this](https://github.com/awlayton/json-schema-2019-09) which we could take inspiration from, not sure if we should rely on the package directly).

FWIW I'm not aware of significant changes between between Draft 07 and Draft 08 that would cause issues for us. This document outlines some of the [incompatible changes](https://json-schema.org/draft/2019-09/release-notes.html) and I don't think we would be impacted by them.

I think it's fine for us to use the Draft 07 TS type for the purposes of doing type checking of common fields (`$id`, `allOf`, etc). Maybe we can add a `@todo` to figure out a better alternative down the line.